Dockerfile.rocm.ubi

```diff
@@ -253,7 +254,7 @@ USER root
 RUN --mount=type=cache,target=/root/.cache/pip \
     --mount=type=cache,target=/root/.cache/uv \
     --mount=type=bind,from=build_vllm,src=/workspace/dist,target=/install/vllm/ \
-    uv pip install /install/vllm/*.whl vllm-tgis-adapter==0.5.3
+    HOME=/root/ uv pip install /install/vllm/*.whl vllm-tgis-adapter==0.5.3
```

Comment:
Not having this caused the final image to have a ~100MB uv cache directory in `/home/vllm/.cache`